test/e2e/templates_test/astparsing/astparsing.go

```diff
@@ -0,0 +1,525 @@
+// Copyright 2024 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package astparsing
+
+import (
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/types"
+	"strings"
+
+	"golang.org/x/tools/go/packages"
+)
+
+const (
+	cobraCommandTypeName             = "*github.com/spf13/cobra.Command"
+	testingTypeName                  = "*testing.T"
+	testPrefix                       = "Test"
+	verifyOutputTemplateMod          = "test"
+	verifyOutputTemplateName         = "VerifyOutputTemplate"
+	verifyOutputTemplateNumberOfArgs = 3
+)
+
+var stringDelimiters = []rune{'`', '\'', '"'}
+
+type CommandBuilderFunc struct {
+	pkg     *packages.Package
+	funDecl *ast.FuncDecl
+}
+
+type CommandBuilderInfo struct {
+	Pkg               *packages.Package
+	CommandOptsStruct *ast.TypeSpec
+	TemplateIdent     *ast.Ident
+	TemplateType      *NamedStructInfo
+	TemplateValue     string
+}
+
+func LoadCommandBuilderInfos(packages []*packages.Package) []*CommandBuilderInfo {
+	commandBuilderFuncDecls := findCommandBuilderFuncDecl(packages)
+
+	builderFuncs := make([]*CommandBuilderInfo, 0)
+
+	for _, commandBuilderFunc := range commandBuilderFuncDecls {
+		fun, err := newCmdBuilderInfo(commandBuilderFunc)
+		if err != nil {
+			fmt.Printf("Error: %v\n", err)
```

Comment:
this func can return []*CommandBuilderInfo and error here, then remove `fmt.Printf("Error: %v\n", err)` and opt for returning the error and propagating error through the functions.

Reply:
Was some left over debugging log, I've removed it.